### ISSUE TITLE
Automatically rerun failed experiment(s) by specifying the number of retries

### ISSUE THREAD

**Issue by yu-iskw** [polyaxon/polyaxon#381]
### Feature motivation
We would like to  rerun failed experiments that are caused by preemption of google cloud, since we are mainly using preemptible instances/GPUs/TPUs of google cloud in order to reduce the cost. So, it is quite natural that experiments are suddenly killed due to the preemption. Especially, when we run a hyper-parameter tuning experiment group, some of the experiments might be killed.  I know we can manually rerun such failed jobs, but it would be great to automatically rerun failed jobs with a certain configuration.

### Feature description
I would like to configure the number of retries of failed experiments. For instance, Apache Airflow has a feature to configure the number of retries when the task is failed.
https://airflow.apache.org/code.html#models

### Feature implementation
For instance, putting `hptuning.retries` can be a possible case to specify the number of retries. 

```
---
version: 1

kind: group

hptuning:
  retries: 1
  concurrency: 5
  random_search:
    n_experiments: 10

  matrix:
    learning_rate:
      linspace: 0.001:0.1:5
    dropout:
      values: [0.25, 0.3]
    activation:
      values: [relu, sigmoid]
```

**Comment by yu-iskw** [comment]
Let me add a request regarding tensorflow. It would be great to restart failed experiments to take over old checkpoints, when the cause to fail the job is preemption. It mean, an experiment failed by preemption restarts with the last check point, rescheduling it on a new launched node.

**Comment by mouradmourafiq** [comment]
This should be possible using the (termination process)[https://github.com/polyaxon/polyaxon/blob/master/core/polyaxon/polyflow/termination/__init__.py#L23], which is a high level mechanism that Polyaxon will inject and reconcile in all created subresources: jobs, deployments, services, distributed jobs (tfjobs, pytorch jobs, mpi jobs) ...

further more, for some use-cases users might use that with the [restart policy](https://github.com/polyaxon/polyaxon/blob/master/core/polyaxon/polyflow/environment/__init__.py#L46)

Fixed in #690 as well.
